api/src/components/publication/controller.ts

```diff
@@ -483,3 +477,53 @@ export const getResearchTopics = async (): Promise<I.JSONResponse> => {
         return response.json(500, { message: 'Unknown server error.' });
     }
 };
+
+export const updateTopics = async (
+    event: I.AuthenticatedAPIRequest<
+        I.UpdatePublicationTopicsRequestBody,
+        undefined,
+        I.UpdatePublicationTopicsPathParams
+    >
+): Promise<I.JSONResponse> => {
+    try {
+        const publication = await publicationService.get(event.pathParameters.id);
+
+        if (!publication) {
+            return response.json(404, {
+                message: 'This publication does not exist.'
+            });
+        }
+
+        const latestVersion = publication.versions.find((version) => version.isLatestVersion);
+
+        if (!latestVersion) {
+            throw Error('Unable to get latest version');
+        }
+
+        if (latestVersion.user.id !== event.user.id) {
+            return response.json(403, {
+                message: 'You do not have permission to update topics for this publication.'
+            });
+        }
+
+        if (latestVersion.currentStatus !== 'DRAFT') {
+            return response.json(400, {
+                message: 'You can not change topics while the publication has no active draft version.'
+            });
+        }
+
+        if (publication.type !== 'PROBLEM') {
+            return response.json(400, {
+                message: 'You can not supply topics for a publication that is not a problem.'
+            });
+        }
+
+        const updateTopics = await publicationService.updateTopics(event.pathParameters.id, event.body.topics);
+
+        return response.json(200, updateTopics);
```

Comment:
This will return a 200 with undefined. Is this intended?

Reply:
No, that wasn't the best way. I've just added some commits that return the publication's topics after a successful update